src/traces/violin/calc.js

```diff
@@ -98,9 +90,17 @@ function calcBandwidth(trace, cdi, vals) {
     // of the distribution.
     // We also want to limit custom bandwidths
     // to not blow up kde computations.
-    return ((cdi.max - cdi.min) / bw) < 1e5 ?
-        bw :
-        (cdi.max - cdi.min) / 100;
+
+    if(trace.bandwidth) {
+        return Math.max(trace.bandwidth, span / 1e4);
```

Comment:
From slack, for the record:

@etpinard: limiting custom bandwidth at (span / 1e5) is still pretty slow. ok if I increase it to span / 1e4 ?

**Me**: sure, that’s still more than 1 bw/px. I’m not quite so worried about it though, at that point the user has asked for something explicitly ridiculous… but I guess we don’t want our mock to take forever to render!